File: src/parsers/txt_parser.py
"""
TXT parser: read plain text file.
"""

from pathlib import Path


def parse_txt(file_path: str) -> str:
    """
    Read plain text file at file_path. Returns content as string (UTF-8).
    """
    path = Path(file_path)
    if not path.exists() or not path.is_file():
        raise FileNotFoundError(f"TXT not found: {file_path}")
    return path.read_text(encoding="utf-8", errors="replace")

File: src/parsers/txt_parser.cpp
#include "document_ingestion/parsers_internal.hpp"
#include <fstream>
#include <sstream>
#include <filesystem>

namespace document_ingestion {

std::string parse_txt(const std::string& file_path) {
  std::filesystem::path p(file_path);
  if (!std::filesystem::exists(p) || !std::filesystem::is_regular_file(p))
    throw std::runtime_error("TXT not found: " + file_path);
  std::ifstream f(file_path);
  if (!f) throw std::runtime_error("Cannot open: " + file_path);
  std::stringstream ss;
  ss << f.rdbuf();
  std::string content = ss.str();
  for (size_t i = 0; i < content.size(); ++i) {
    if (static_cast<unsigned char>(content[i]) >= 0x80 && 
        static_cast<unsigned char>(content[i]) < 0xC0) {
      content[i] = '?';
    }
  }
  return content;
}

}  // namespace document_ingestion
